var/spack/repos/builtin/packages/arpack-ng/package.py

```diff
@@ -129,19 +128,16 @@ def install(self, spec, prefix):
                 'F77=%s' % spec['mpi'].mpif77
             ])
 
-        if '+shared' in spec:
-            options.extend([
-                '--with-blas=%s' % to_link_flags(
-                    spec['blas'].blas_shared_lib),
-                '--with-lapack=%s' % to_link_flags(
-                    spec['lapack'].lapack_shared_lib)
-            ])
-        else:
-            options.extend([
-                '--with-blas=%s' % spec['blas'].blas_static_lib,
-                '--with-lapack=%s' % spec['lapack'].lapack_static_lib,
-                '--enable-shared=no'
-            ])
+        options.extend([
+                '--with-blas={0}'.format(
+                    ' '.join(spec['blas'].blas_libs.ld_flags)
```

Comment:
Before anybody says anything : since `' '.join(ld_flags)` seems a common idiom I'll modify the PR to avoid having to write the join in the package :smile: